// @reportportal/agent-js-webdriverio/build/constants/fileTypes.js
"use strict";
/*
 *  Copyright 2021 EPAM Systems
 *
 *  Licensed under the Apache License, Version 2.0 (the "License");
 *  you may not use this file except in compliance with the License.
 *  You may obtain a copy of the License at
 *
 *  http://www.apache.org/licenses/LICENSE-2.0
 *
 *  Unless required by applicable law or agreed to in writing, software
 *  distributed under the License is distributed on an "AS IS" BASIS,
 *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 *  See the License for the specific language governing permissions and
 *  limitations under the License.
 *
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.FILE_TYPES = void 0;
var FILE_TYPES;
(function (FILE_TYPES) {
    FILE_TYPES["XML"] = "application/xml";
    FILE_TYPES["HTML"] = "application/html";
    FILE_TYPES["JAVASCRIPT"] = "application/javascript";
    FILE_TYPES["JSON"] = "application/json";
    FILE_TYPES["PHP"] = "application/php";
    FILE_TYPES["CSS"] = "application/css";
    FILE_TYPES["TEXT"] = "text/plain";
    FILE_TYPES["PNG"] = "image/png";
    FILE_TYPES["JPG"] = "image/jpg";
})(FILE_TYPES = exports.FILE_TYPES || (exports.FILE_TYPES = {}));
