// @reportportal/agent-js-webdriverio/build/constants/statuses.js
"use strict";
/*
 *  Copyright 2020 EPAM Systems
 *
 *  Licensed under the Apache License, Version 2.0 (the "License");
 *  you may not use this file except in compliance with the License.
 *  You may obtain a copy of the License at
 *
 *  http://www.apache.org/licenses/LICENSE-2.0
 *
 *  Unless required by applicable law or agreed to in writing, software
 *  distributed under the License is distributed on an "AS IS" BASIS,
 *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 *  See the License for the specific language governing permissions and
 *  limitations under the License.
 *
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.RP_STATUSES = void 0;
var RP_STATUSES;
(function (RP_STATUSES) {
    RP_STATUSES["FAILED"] = "failed";
    RP_STATUSES["PASSED"] = "passed";
    RP_STATUSES["SKIPPED"] = "skipped";
    RP_STATUSES["STOPPED"] = "stopped";
    RP_STATUSES["INTERRUPTED"] = "interrupted";
    RP_STATUSES["CANCELLED"] = "cancelled";
    RP_STATUSES["INFO"] = "info";
    RP_STATUSES["WARN"] = "warn";
})(RP_STATUSES = exports.RP_STATUSES || (exports.RP_STATUSES = {}));
