// @reportportal/agent-js-webdriverio/build/constants/testItemTypes.js
"use strict";
/*
 *  Copyright 2021 EPAM Systems
 *
 *  Licensed under the Apache License, Version 2.0 (the "License");
 *  you may not use this file except in compliance with the License.
 *  You may obtain a copy of the License at
 *
 *  http://www.apache.org/licenses/LICENSE-2.0
 *
 *  Unless required by applicable law or agreed to in writing, software
 *  distributed under the License is distributed on an "AS IS" BASIS,
 *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 *  See the License for the specific language governing permissions and
 *  limitations under the License.
 *
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.CUCUMBER_TYPE = exports.TYPES = void 0;
var TYPES;
(function (TYPES) {
    TYPES["SUITE"] = "SUITE";
    TYPES["STEP"] = "STEP";
    TYPES["TEST"] = "TEST";
})(TYPES = exports.TYPES || (exports.TYPES = {}));
var CUCUMBER_TYPE;
(function (CUCUMBER_TYPE) {
    CUCUMBER_TYPE["FEATURE"] = "feature";
    CUCUMBER_TYPE["SCENARIO"] = "scenario";
})(CUCUMBER_TYPE = exports.CUCUMBER_TYPE || (exports.CUCUMBER_TYPE = {}));
